### PR TITLE
Added 3 messages with links to TestCafe Studio

### PR DESCRIPTION
**Pull request by miherlosev**
The new message with link to TestCafe Studio is displayed in the CLI interface after each 20 runs.
Now, there are 3 such messages.

The texts will be reviewed by Rey. But we check them now and apply the changes later if it will be necessary.

### CODE REVIEW

**Comment by miherlosev** [comment]
FPR @kirovboris @AndreyBelym @VasilyStrelyaev

**Comment by VasilyStrelyaev** [comment]
https://github.com/DevExpress/testcafe/pull/3733/files#diff-93d60a2037f34104929d4d77560fb4c3R32

So, these warning are for our own debug purposes? Will they be displayed only with the `--dev` flag?

**Comment by miherlosev** [comment]
> So, these warning are for our own debug purposes? Will they be displayed only with the --dev flag?

* It's internal debug messages. 
* They will display if you specify `DEBUG` environment variable (for more detail see https://www.npmjs.com/package/debug#windows-command-prompt-notes).

**Comment by VasilyStrelyaev** [comment]
Okay, then I guess we don't have to run them through the correctors. I'll just make one edit.
The promotional messages are yet to be approved by Ray.

File: src/notifications/warning-message.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ export default {
 
     problematicPathPatternPlaceholderForVideoRecording: 'The {placeholderList} path pattern placeholder{suffix} cannot be applied to the recorded video.\n' +
                                                         '\n' +
-                                                        'The placeholder{suffix} {verb} replaced with an empty string.'
+                                                        'The placeholder{suffix} {verb} replaced with an empty string.',
+
+    cannotLoadMarketingData:         'An error has occurred while reading the marketing data. Error details:\n\n{err}',
+    cannotSaveMarketingData:         'An error has occurred while reading the marketing data. Error details:\n\n{err}',
+    cannotCalculateMarketingMessage: 'Cannot calculate valid marketing message. Displayed message index is {index}'
```
Comment (VasilyStrelyaev) [inline src/notifications/warning-message.js]:
```suggestion
    cannotCalculateMarketingMessage: 'Cannot determine which promotional message to display. Attempted to display a message no. {index}'
```

File: src/notifications/warning-message.js [hunk 2]
```diff
@@ -27,6 +27,10 @@ export default {
 
     problematicPathPatternPlaceholderForVideoRecording: 'The {placeholderList} path pattern placeholder{suffix} cannot be applied to the recorded video.\n' +
                                                         '\n' +
-                                                        'The placeholder{suffix} {verb} replaced with an empty string.'
+                                                        'The placeholder{suffix} {verb} replaced with an empty string.',
+
+    cannotLoadMarketingData:         'An error has occurred while reading the marketing data. Error details:\n\n{err}',
+    cannotSaveMarketingData:         'An error has occurred while reading the marketing data. Error details:\n\n{err}',
```
Comment (VasilyStrelyaev) [inline src/notifications/warning-message.js]: Maybe `saving` - according to the error ID

```suggestion
    cannotSaveMarketingData:         'An error has occurred while saving the marketing data. Error details:\n\n{err}',
```

File: src/marketing/messages.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const DOMAIN_1 = 'https://devexpress.com/testcafe-studio';
+const DOMAIN_2 = 'https://devexpress.com/test-cafe-studio';
+const DOMAIN_3 = 'https://devexpress.com/testcafe/studio';
+
+const HOW_TO_IMPROVE_TESTING_EXPERIENCE = `Learn how to improve your testing experience with TestCafe Studio edition - ${DOMAIN_1}`;
```
Comment (kirovboris) [inline src/marketing/messages.js]: Let's add an empty line before this messages, it doesn't look good now.
image

File: src/marketing/index.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import DataFile from './data-file';
+import MESSAGES from './messages';
+import renderTemplate from '../utils/render-template';
+import WARNING_MESSAGES from '../notifications/warning-message';
+import log from '../cli/log';
+import debug from 'debug';
+
+export const NUMBER_RUNS_BETWEEN_SHOW_MESSAGE = 20;
```
Comment (kirovboris) [inline src/marketing/index.js]: So, we discussed and decided to show message on first run(in other words we show it if the DataFile is not exists) for this version and every 10 runs.

File: src/cli/cli.js [hunk 5]
```diff
@@ -100,6 +100,8 @@ async function runTests (argParser) {
 
     try {
         failed = await runner.run(opts);
+
+        await marketing.showMessageWithLinkToTestCafeStudio();
```
Comment (kirovboris) [inline src/cli/cli.js]: Let's not to write this message in stdout for parsable reporters like xunit, json etc.
So I suggest to output these messages only for 'spec', 'list', 'minimal' reporters.

**Comment by kirovboris** [comment]
We decided to show on first run as well. LGTM in other.

**Pull request merged**